adriconf/HelpersWayland.cpp

```diff
@@ -18,147 +17,36 @@ bool HelpersWayland::hasProperLibEGL() {
-
-    ok = eglInitialize(display, &ignore, &ignore);
-    if (!ok)
-        std::cerr << _("eglInitialize() failed\n");
-
-    EGLint configs_size = 256;
-    EGLConfig* configs = new EGLConfig[configs_size];
-    EGLint num_configs;
-    ok = eglChooseConfig(
-        display,
-        NULL,
-        configs,
-        configs_size, // num requested configs
-        &num_configs); // num returned configs
-    if (!ok)
-        std::cerr << _("eglChooseConfig() failed\n");
-    if (num_configs == 0)
-        std::cerr << _("failed to find suitable EGLConfig\n");
-    EGLConfig config = configs[0];
-    delete [] configs;
-
-    EGLContext context = eglCreateContext(
-        display,
-        config,
-        EGL_NO_CONTEXT,
-        NULL);
-    if (!context)
-        std::cerr << _("eglCreateContext() failed\n");
-
-    EGLSurface surface = eglCreateWindowSurface(
-        display,
-        config,
-        native_window,
-        NULL);
-    if (!surface)
-        std::cerr << _("eglCreateWindowSurface() failed\n");
-
-    ok = eglMakeCurrent(display, surface, surface, context);
-    if (!ok)
-    std::cerr << _("eglMakeCurrent() failed\n");
-
-    *out_display = display;
-    *out_config = config;
-    *out_context = context;
-    *out_window_surface = surface;
-}
-
 const char *HelpersWayland::queryDriverName() {
-    const char *x_display_name = NULL;
-    Display *x_display;
-    Window x_window;
-
-    const char *ret;
-    setup_x(x_display_name,
-            &x_display,
-            &x_window);
-
     EGLDisplay egl_display;
-    EGLConfig egl_config;
-    EGLContext egl_context;
-    EGLSurface egl_surface;
-    setup_egl(EGL_OPENGL_API,
-              x_display,
-              x_window,
-              &egl_display,
-              &egl_config,
-              &egl_context,
-              &egl_surface);
+    EGLint egl_major, egl_minor;
+    const char *driver_name;
 
+    egl_display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+
+    if (!eglInitialize(egl_display, &egl_major, &egl_minor)) {
+        std::cerr << _("eglInitialize() failed\n");
+    }
 
     driverName = (eglGetDisplayDriverName_t *) eglGetProcAddress("eglGetDisplayDriverName");
-    ret = (* driverName) (egl_display);
+    driver_name = (* driverName) (egl_display);
 
-    XCloseDisplay(x_display);
-    return ret;
+    return driver_name;
 }
 
 const char *HelpersWayland::queryDriverConfig() {
-    const char *x_display_name = NULL;
-    Display *x_display;
-    Window x_window;
-    setup_x(x_display_name,
-            &x_display,
-            &x_window);
-
     EGLDisplay egl_display;
-    EGLConfig egl_config;
-    EGLContext egl_context;
-    EGLSurface egl_surface;
-    setup_egl(EGL_OPENGL_API,
-              x_display,
-              x_window,
-              &egl_display,
-              &egl_config,
-              &egl_context,
-              &egl_surface);
+    EGLint egl_major, egl_minor;
 
+    egl_display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
```

Comment:
This has the same issue as the XWayland implementation: only the first screen/driver is queried.
What happens in multi-screen/multi-gpu setups?

Reply:
To make this more clear: I also have no idea on how to solve this.
I started to investigate QT libraries, as they seem to offer some interesting options like screen count in one of their global objects but didn't digged further.
As this is not making things worse, we can leave this as-is for now.